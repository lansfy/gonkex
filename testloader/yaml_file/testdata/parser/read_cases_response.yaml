- name: loader MUST substitute responseArgs to response components
  method: GET
  path: /some/path
  response:
    200: '{"hello": {{ .foo }} }'
    400: '{"hello": {{ .foo }} }'
    500: '{"hello": {{ .foo }} }'
  responseHeaders:
    200:
      X-Header: '{{ .foo }}'
    400:
      X-Header: '{{ .foo }}'
    500:
      X-Header: '{{ .foo }}'
  cases:
    - responseArgs:
        200:
          foo: "value1"
        400:
          foo: "value2"
    - responseArgs:
        200:
          foo: "value3"
        400:
          foo: "value4"
