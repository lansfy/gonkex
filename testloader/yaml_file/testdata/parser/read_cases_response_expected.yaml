- GetName: "loader MUST substitute responseArgs to response components #1"
  GetMethod: GET
  Path: /some/path
  GetResponses:
    200: '{"hello": value1 }'
    400: '{"hello": value2 }'
    500: '{"hello": {{ .foo }} }' # not part of cases
  GetResponseHeaders:
    200:
      X-Header: 'value1'
    400:
      X-Header: 'value2'
    500:
      X-Header: '{{ .foo }}' # not part of cases
  GetFileName: testdata/parser/read_cases_response.yaml
  OneOfCase: true
  FirstTestInFile: true

- GetName: "loader MUST substitute responseArgs to response components #2"
  GetMethod: GET
  Path: /some/path
  GetResponses:
    200: '{"hello": value3 }'
    400: '{"hello": value4 }'
    500: '{"hello": {{ .foo }} }' # not part of cases
  GetResponseHeaders:
    200:
      X-Header: 'value3'
    400:
      X-Header: 'value4'
    500:
      X-Header: '{{ .foo }}' # not part of cases
  GetFileName: testdata/parser/read_cases_response.yaml
  OneOfCase: true
  LastTestInFile: true
